mlflow/tracking/fluent.py

```diff
@@ -1342,32 +1332,12 @@ def _get_experiment_id_from_env():
 
 
 def _get_experiment_id():
-    # TODO: Replace with None for 1.0, leaving for 0.9.1 release backcompat with existing servers
-    deprecated_default_exp_id = "0"
-
     return (
         _active_experiment_id
         or _get_experiment_id_from_env()
-        or (is_in_databricks_notebook() and get_notebook_id())
-        or (is_in_databricks_job() and get_job_type_info() == "NORMAL" and _create_job_experiment())
-    ) or deprecated_default_exp_id
-
-
-def _create_job_experiment() -> str:
-    job_id = get_job_id()
-    tags = {}
-    tags[MLFLOW_DATABRICKS_JOB_TYPE_INFO] = get_job_type_info()
-    tags[MLFLOW_EXPERIMENT_SOURCE_TYPE] = "JOB"
-    tags[MLFLOW_EXPERIMENT_SOURCE_ID] = job_id
-
-    experiment_id = create_experiment(get_experiment_name_from_job_id(job_id), None, tags)
-    _logger.debug(
-        "Job experiment with experiment_id '%s' created",
-        experiment_id,
+        or default_experiment_registry.get_experiment_id()
```

Comment:
`default_experiment_registry.get_experiment_id()` may be a resource-intensive or long-running operation. For example, the `DatabricksJobExperimentProvider` makes an API request every time this method is called. 

Can we set `_active_experiment_id` to the result of `default_experiment_registry.get_experiment_id()` to avoid, e.g., making a `CreateExperiment` API call every time `_get_experiment_id()` is invoked?

Reply:
Edit: It's important to let users make changes to environment variable configs and have them be reflected. Instead, of my suggestion above, can we cache within `DatabricksJobExperimentProvider`?